types/react-datepicker/index.d.ts

```diff
@@ -19,109 +19,113 @@ export function registerLocale(localeName: string, localeData: {}): void;
-	onChangeRaw?(event: React.FocusEvent<HTMLInputElement>): void;
-	onClickOutside?(event: React.MouseEvent<HTMLDivElement>): void;
-	onFocus?(event: React.FocusEvent<HTMLInputElement>): void;
-	onInputClick?(): void;
-	onInputError?(err: {code: number; msg: string}): void;
-	onKeyDown?(event: React.KeyboardEvent<HTMLDivElement>): void;
-	onMonthChange?(date: Date): void;
-	onSelect?(date: Date, event: React.SyntheticEvent<any> | undefined): void;
-	onWeekSelect?(firstDayOfWeek: Date, weekNumber: string | number, event: React.SyntheticEvent<any> | undefined): void;
-	onYearChange?(date: Date): void;
-	open?: boolean;
-	openToDate?: Date;
-	peekNextMonth?: boolean;
-	placeholderText?: string;
-	popperClassName?: string;
-	popperContainer?(props: { children: React.ReactNode[] }): React.ReactNode;
-	popperModifiers?: Popper.Modifiers;
-	popperPlacement?: string;
-	popperProps?: {};
-	preventOpenOnFocus?: boolean;
-	previousMonthButtonLabel?: string;
-	readOnly?: boolean;
-	renderCustomHeader?(params: {
-		date: Date;
-		changeYear(year: number): void;
-		changeMonth(month: number): void;
-		decreaseMonth(): void;
-		increaseMonth(): void;
-		prevMonthButtonDisabled: boolean;
-		nextMonthButtonDisabled: boolean;
-	}): React.ReactNode;
-	renderDayContents?(dayOfMonth: number): React.ReactNode;
-	required?: boolean;
-	scrollableMonthYearDropdown?: boolean;
-	scrollableYearDropdown?: boolean;
-	selected?: Date | null;
-	selectsEnd?: boolean;
-	selectsStart?: boolean;
-	shouldCloseOnSelect?: boolean;
-	showDisabledMonthNavigation?: boolean;
-	showMonthDropdown?: boolean;
-	showMonthYearDropdown?: boolean;
-	showTimeSelect?: boolean;
-	showTimeSelectOnly?: boolean;
-	showWeekNumbers?: boolean;
-	showYearDropdown?: boolean;
-	startDate?: Date;
-	startOpen?: boolean;
-	tabIndex?: number;
-	timeCaption?: string;
-	timeFormat?: string;
-	timeIntervals?: number;
-	title?: string;
-	todayButton?: React.ReactNode;
-	useShortMonthInDropdown?: boolean;
-	useWeekdaysShort?: boolean;
-	value?: string;
-	weekLabel?: string;
-	withPortal?: boolean;
-	yearDropdownItemNumber?: number;
+  adjustDateOnChange?: boolean;
+  allowSameDay?: boolean;
+  autoComplete?: string;
+  autoFocus?: boolean;
+  calendarClassName?: string;
+  calendarContainer?(props: { children: React.ReactNode[] }): React.ReactNode;
+  children?: React.ReactNode;
+  className?: string;
+  clearButtonTitle?: string;
+  customInput?: React.ReactNode;
+  customInputRef?: string;
+  dateFormat?: string | string[];
+  dateFormatCalendar?: string;
+  dayClassName?(date: Date): string | null;
+  disabled?: boolean;
+  disabledKeyboardNavigation?: boolean;
+  dropdownMode?: "scroll" | "select";
+  endDate?: Date;
+  excludeDates?: Date[];
+  excludeTimes?: Date[];
+  filterDate?(date: Date): boolean;
+  fixedHeight?: boolean;
+  forceShowMonthNavigation?: boolean;
+  formatWeekDay?(date: Date): string;
+  formatWeekNumber?(date: Date): string | number;
+  highlightDates?: HighlightDates[];
```

Comment:
I think this should be 
```suggestion
  highlightDates?: (Date|HighlightDates)[];
```
because it can be a date or an object.  See [this function](https://github.com/Hacker0x01/react-datepicker/blob/master/src/date_utils.js#L383-L415).